libexec/halyard.sh

```diff
@@ -123,23 +122,51 @@ load() {
   display "${target[@]}"
 }
 
-run() {
-  # Ensure Docker Desktop is up
+# Starts Docker if not already running
+docker_start() {
   open --background -a Docker &&
     if ! docker system info >/dev/null 2>&1; then
       echo "Staring Docker..." &&
         while ! docker system info >/dev/null 2>&1; do
           sleep 1
         done
     fi
+}
+
+# Runs Memcheck in a Docker container instance
+# with the loaded files
+docker_run() {
+  local run_with_make="$1"
+  local files=("${@:2}")
+
+  # TODO: Redirect output, parse, and display for user
+  # Runs a full leak check and displays results
+  if [ "$run_with_make" = true ]; then
+    printf "\n${HALYARD_SAYS} makefile detected\n"
+    printf "${HALYARD_SAYS} executable path: "; read exec_path; printf "\n"
+    docker run --rm -ti -v $PWD:/test halyard:0.1 bash -c \
+      "cd /test/; 
+       echo 'making ${exec_path}... ';
+       make && valgrind --leak-check=full ./${exec_path}"
+  else
+    docker run --rm -ti -v $PWD:/test halyard:0.1 bash -c \
+      "cd /test/; 
+       $compiler -o memcheck ${files[*]} &&
+       valgrind --leak-check=full ./memcheck"
+  fi
+}
+
+run() {
+  docker_start
 
   # Array for source files.
   local target=()
   local extension
   local compiler
   local file_count=0
+  local make_detected=false
 
-  for file in "$CONTAINER_PATH"/*; do
+  for file in "${CONTAINER_PATH}"/*; do
     extension="${file##*.}"
     # Set compiler based on source extension
     if [ $extension = "c" ] || [ $extension = "cpp" ] || [ $extension = "cc" ]; then
```

Comment:
I truncated the names to get the file extension for the display; in doing so I noted two things:

* if a file name doesn't have an extension after the `.` then whatever is `"${file##*.}"` is considered
* and that here in `<line 172>` /dir1/dir2/filename will be compared to "c" or "cpp"

I don't think this is a deal at all. Just something I noticed. Since we already have  `"${file##*.}"` there's no use in trying to truncate it to find a match since it already won't match and the `##*.` did the work.

Reply:
Yeah, and the idea still works, as long as everything before the filename is truncated. 

The problem is that using `"${file##*.}"` assumes there are no other `.` in the path,  but `.halyard` makes this an incorrect assumption. 

Truncating everything before the filename fixes this problem, and still allows the files to be properly displayed on screen.  Also this truncate is only for this `display()` function.  It will not carry over into `run()`